atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionConflictDetectionManager.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.annotation.Nullable;
+
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.transaction.api.ConflictHandler;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+
+public final class TransactionConflictDetectionManager {
+
+    private final ConflictDetectionManager delegate;
+    private final Map<TableReference, Optional<ConflictHandler>> conflictHandlers = new ConcurrentHashMap<>();
+
+    public TransactionConflictDetectionManager(ConflictDetectionManager delegate) {
+        this.delegate = delegate;
+    }
+
+    public void disableReadWriteConflict(TableReference tableRef) {
+        conflictHandlers.compute(tableRef, (tableReference, nullableCurrentValue) -> {
+            ConflictHandler conflictHandler = delegate.get(tableReference);
```

Comment:
Surprise, surprise: guava's CacheLoader does not allow null as a value, so this will never actually return a null.